src/components/Draggable/FilterDraggable.tsx

```diff
@@ -0,0 +1,219 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import React, { useState } from 'react'
+import { useDrag } from 'react-dnd'
+import { FixedSizeList } from 'react-window'
+import { Button, Checkbox, Dropdown, DropdownItem, HFlow, Icon, TextField, Tooltip } from '..'
+import { useTheme } from '../../styles'
+import { DraggableProps } from './Draggable'
+
+export interface FilterDraggableProps<T> extends NonNullable<DraggableProps<T>> {
+  styles: any
+}
+
+export function FilterDraggable<T>(props: FilterDraggableProps<T>) {
+  const {
+    name,
+    type,
+    origin,
+    value,
+    filterValues,
+    filterState,
+    onDragEnd,
+    handleFilterUpdate,
+    formatter,
+    onKeyNav,
+    styles,
+  } = props
+
+  const [searchedFilterSet, setSearchedFilterSet] = useState<Array<string>>(filterValues)
+
+  const [open, setOpen] = useState(false)
+
+  const [all, setAll] = useState<0 | 1 | 2>(
```

Comment:
Those states (0, 1, 2) could be replaced to string so it'll have a name more easily to understand.